src/storage/recovery/recovery_manager.cpp

```diff
@@ -41,15 +43,26 @@ void RecoveryManager::WaitForRecoveryToFinish() {
 
 void RecoveryManager::RecoverFromLogs() {
   // Replay logs until the log provider no longer gives us logs
+  int exhausted_timer = 0;
   while (true) {
+    //STORAGE_LOG_INFO("[Recovery Manager]: Grabbing next log record...");
     auto pair = log_provider_->GetNextRecord();
     auto *log_record = pair.first;
 
     // If we have exhausted all the logs, break from the loop
-    if (log_record == nullptr) break;
+    if (log_record == nullptr) {
+      STORAGE_LOG_INFO("[Recovery Manager]: Exhausted all logs!");
+      if (exhausted_timer > 3) {
```

Comment:
What is the purpose of this timer?

Reply:
This is a hack to keep the recovery task keep running. I am not yet shutting it down correctly.